src/lib/nvme_pt_ioctl.c

```diff
@@ -727,6 +733,94 @@ static int validate_sessn(struct opal_device *dev)
 	return 0;
 }
 
+static int  parse_tper_host_prop(struct sed_device *device)
+{
+	struct opal_device *dev = device->priv;
+	int payload_len = dev->payload.len;
+	struct sed_tper_properties *tper = &device->discv.sed_tper_props;
+	uint64_t val_ptr = sizeof(struct sed_tper_properties);
+	uint64_t val, data_ptr = 0;
+
+	for (int i = 0; i < payload_len; i++) {
+		if(resp_token_match(dev->payload.tokens[i], OPAL_STARTNAME)) {
+			val = dev->payload.tokens[i+2]->vals.uint;
+
+			if (data_ptr < val_ptr)  {
+				memcpy((uint8_t *)tper + data_ptr, &val, sizeof(uint64_t));
+				data_ptr += sizeof(uint64_t);
+			}
+		}
+	}
+
+	return 0;
+}
+
+static struct opal_req_item opal_tper_host_prep_cmd[] = {
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_STARTNAME } },
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_HOSTPROPERTIES } },
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_STARTLIST } },
+
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_STARTNAME } },
+	{ .type = OPAL_BYTES, .len = 16, .val = { .bytes = (uint8_t *)"MaxComPacketSize" } },
+	{ .type = OPAL_U64, .len = 2, .val = { .uint = 2048 } },
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_ENDNAME } },
+
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_STARTNAME } },
+	{ .type = OPAL_BYTES, .len = 24, .val = { .bytes = (uint8_t *)"MaxResponseComPacketSize" } },
+	{ .type = OPAL_U64, .len = 2, .val = { .uint = 4096 } },
```

Comment:
is this value correct?

Reply:
1st : the max response com packet size should be changed to 2048 here.
2nd : These values are derived from the Section 3.2.1.2 of Opal SSC Application note.
the spec takes 4k buffer as an example. However, we consider 2k as our buffer size

Reply:
Thx. I think these values should match whatever sedcli is currently capable of, for example if sedcli can send packets at max 2048B in size we should advertise this to Tper.